versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/spi/TryLoopState.java

```diff
@@ -17,51 +17,73 @@
 
 import java.util.concurrent.ThreadLocalRandom;
 import java.util.concurrent.TimeUnit;
-import java.util.function.Supplier;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
 import org.projectnessie.versioned.Hash;
 import org.projectnessie.versioned.ReferenceRetryFailureException;
 import org.projectnessie.versioned.persist.adapter.DatabaseAdapterConfig;
 
 /** Retry-logic for attempts for compare-and-swap-like operations. */
 public final class TryLoopState implements AutoCloseable {
 
-  static final int INITIAL_LOWER_BOUND = 5;
-  static final int INITIAL_UPPER_BOUND = 25;
-
   private final MonotonicClock monotonicClock;
   private final long t0;
   private final long maxTime;
   private final int maxRetries;
+  private final Function<TryLoopState, String> retryErrorMessage;
+  private final BiConsumer<Boolean, TryLoopState> completionNotifier;
+  private final long maxSleep;
+  private long lowerBound;
+  private long upperBound;
   private int retries;
-  private final Supplier<String> retryErrorMessage;
-
-  private long lowerBound = INITIAL_LOWER_BOUND;
-  private long upperBound = INITIAL_UPPER_BOUND;
 
   TryLoopState(
-      Supplier<String> retryErrorMessage,
+      Function<TryLoopState, String> retryErrorMessage,
       DatabaseAdapterConfig config,
-      MonotonicClock monotonicClock) {
+      MonotonicClock monotonicClock,
+      BiConsumer<Boolean, TryLoopState> completionNotifier) {
     this.retryErrorMessage = retryErrorMessage;
     this.maxTime = TimeUnit.MILLISECONDS.toNanos(config.getCommitTimeout());
     this.maxRetries = config.getCommitRetries();
     this.monotonicClock = monotonicClock;
     this.t0 = monotonicClock.currentNanos();
+    this.lowerBound = config.getRetryInitialSleepMillisLower();
+    this.upperBound = config.getRetryInitialSleepMillisUpper();
+    this.maxSleep = config.getRetryMaxSleepMillis();
```

Comment:
should the field names include the `Millis` suffix? because for example `maxTime` above is in nanos.